zerver/webhooks/newrelic/doc.md

```diff
@@ -1,27 +1,121 @@
 
+{start_tabs}
+
 1. {!create-channel.md!}
 
 1. {!create-an-incoming-webhook.md!}
 
-1. {!generate-integration-url.md!}
+1. {!generate-webhook-url-basic.md!}
 
-1. On [New Relic](https://one.newrelic.com),
-  select **Alerts & AI**.
+1. Go to the [**Destinations**][destinations] menu and select **Webhook** in
+   the **Add a destination** section. ([*Official Guide*][official-guide])
 
-1. Navigate to **Notification channels**.
+1. Set **Endpoint URL** to the URL you generated.
 
-1. Create a new notification channel. Select channel type of **Webhook**, choose a name (e.g., "Zulip"), enter the webhook url created earlier as **Base Url**.
+1. Go to the [**Workflows**][workflows] menu and click on **+ Add a Workflow**
+   button. After filtering the trigger condition, select on the **Webhook**
+   option in the **Add Channel** section. ([*Workflow doc*][workflow-doc])
 
-1. It should look like:
-  ![](/static/images/integrations/newrelic/newrelic.png)
+1. Once you're in the **Edit notification message** menu, select the
+   destination for Zulip you've just created earlier.
 
-1. The webhook works with the default payload, click **Create channel**.
+1. If you want to include additional information to the notification, refer
+   to this [section](#displaying-custom-fields). There's also an
+   optional [base payload template](#recommended-base-payload-template) we
+   recommend you to use.
 
-1. After creating the channel send a test notification to make sure it works.
+1. Select **Save message** and the **Activate Workflow** to finish.
+
+{end_tabs}
 
 {!congrats.md!}
 
 ![](/static/images/integrations/newrelic/001.png)
+
 ![](/static/images/integrations/newrelic/002.png)
-![](/static/images/integrations/newrelic/003.png)
+
+### Displaying Custom Fields
+
+The New Relic integration supports the [custom payload feature][custom-payload]
+offered by New Relic. To include your custom fields in your Zulip New Relic
+notification, follow these rules:
+
+* Put all your custom key-value pairs in a new dictionary called
+  `zulipCustomFields`.
+
+* The keys inside the `zulipCustomFields` dict will be displayed
+  in the notification message as is, so we recommend you to use
+  descriptive and human-readable words for the keys.
+
+* The supported data-type for the custom payload are: `string`, `list`, `int`
+  and `bool`. Custom payload containing `dict` are not supported.
+
+  *example payload with custom fields:*
+
+       {
+         {% raw %}
+         "id": {{ json issueId }},
+
+         ...
+
+         "zulipCustomFields": {
+            "Your Custom Payload": "somedata123",
+            "Storage Hardware": ["SSD", "HDD],
+            "Closed Count": {{ json closedIncidentsCount }}
+         }
+         {% endraw %}
+      }
+
+  *the Zulip notification from the payload above:*
+  ![](/static/images/integrations/newrelic/custom_payload_notif.png)
+
+Only custom fields inside the `zulipCustomFields` dictionary will be displayed.
```

Comment:
I'm not sure why this is needed?

Reply:
Hmm yeah, on second thought the whole section implies just that.